### DISCUSSION TITLE
How to execute the sql file in Duckdb

### DISCUSSION THREAD

**Discussion by woshimajintao** [duckdb/duckdb#9423] [Q&A]
Now I have a sql file with 99queries.
I wonder how to execute the sql file in Duckdb.
I means to execute the whole sql file.

**Comment by Tishj** [comment]
`test.sql`:
```sql
create table tbl (a integer, b bool);
insert into tbl values (1, true), (2, false);
```

`duckdb -init test.sql`
```sql
-- Loading resources from test.sql
Enter ".help" for usage hints.
D select * from tbl;
┌───────┬─────────┐
│   a   │    b    │
│ int32 │ boolean │
├───────┼─────────┤
│     1 │ true    │
│     2 │ false   │
└───────┴─────────┘
```

**Reply by woshimajintao** [reply]
Thanks!And i also have a question. I wonder how can I get the runtime of the SQL file.

**Reply by Tishj** [reply]
`time duckdb < test.sql`

**Reply by Tishj** [reply]
Also I'm gonna ask that you stop making new discussions, your questions are very specific and all seem related, stick to one thread for your inquiries please

**Reply by woshimajintao** [reply]
Thanks sir.Because I am a beginner of the tool.I donot understand it.

**Reply by woshimajintao** [reply]
Dear sir, could you please give me some suggestions about optimizing about TPC-DS queries of the Duckdb?

**Reply by szarnyasg** [reply]
@woshimajintao We are unable to help with such requests on GitHub Discussions. Optimizing TPC-DS queries is a complex task. You'll have to run a large number of experiments and try various optimizations.

**Reply by woshimajintao** [reply]
Thanks for your reply.Could you please recommond some papers or blogs related to the optimizion about about TPC-DS queries of the Duckdb?
